Shared/Actions/ArchiveParams.cs

```diff
@@ -83,19 +101,33 @@ public ArchiveParams Prefixes(List<string> prefixes)
             return this;
         }
 
+        /// <inheritdoc />
         /// <summary>
         /// Validate object model
         /// </summary>
         public override void Check()
         {
-            if ((m_publicIds == null || m_publicIds.Count == 0) &&
-                (m_prefixes == null || m_prefixes.Count == 0) &&
-                (m_tags == null || m_tags.Count == 0))
-                throw new ArgumentException("At least one of the following \"filtering\" parameters needs to be specified: PublicIds, Tags or Prefixes.");
+            if (m_publicIds?.Any() != true &&
+                m_fullyQualifiedPublicIds?.Any() != true &&
+                m_prefixes?.Any() != true &&
+                m_tags?.Any() != true)
+                throw new ArgumentException("At least one of the following \"filtering\" parameters needs " +
+                                            "to be specified: PublicIds, FullyQualifiedPublicIds, Tags or Prefixes.");
+
+            if (m_resourceType == "auto")
```

Comment:
@const-cloudinary maybe:
if (!(m_resourceType=="auto" ^  m_fullyQualifiedPublicIds?.Any())) throw new ArgumentException ("Archiving assets from different resource types requires both fully qualified names and setting ResourceType to \"auto\" ")

Reply:
To create an archive with multiple types of assets, you must set ResourceType to 'auto' and provide fully-qualified public IDs (For example, 'video/upload/myvideo').